### ISSUE TITLE
create mdmctl add device -serial SERIAL - concept for placeholders

### ISSUE THREAD

**Issue by erikng** [micromdm/micromdm#483]
### What version of `micromdm` are you using?
1.3.1

What I would like to see is the ability to add device records into micromdm (say ones that aren't in DEP).

The idea here is I want to create records for all known company devices and then see if there is a way to blacklist every other device. This would mean that unless a device record exists for a particular serial, during enrollment you will always get a 401 and cannot fully enrolled into the MDM server.

`mdmctl add device -serial SERIAL` could be looped through a list (or entered as a CSV) and mdmctl would create records for these devices if they dont exist with a unique UDID.

@bruienne had an idea for using `depsim` to simulate the records and then import them, and while that is an interesting workaround, would be nice to have this built in.

**Comment by jessepeterson** [comment]
An enrollment whitelist (as opposed to black list with the current `block` feature). Interesting!

**Comment by erikng** [comment]
yeah, I think the whitelist would be all known devices in your db and you could still use the blacklist to block devices that were stolen/went missing/etc.

If someone wanted to utilize this feature and was already in production, they could use `get devices` and compare with their inventory system and remove any device records they didn't want to allow.

maybe it's a mixture of this command and `microdm -onlyallowregistereddevices` or something like that.

**Comment by knightsc** [comment]
Adding links to some related ideas that have been discussed in the past: https://github.com/micromdm/micromdm/issues/310 https://github.com/micromdm/micromdm/issues/320
